Snake and Ladder Problem.md

```diff
@@ -0,0 +1,121 @@
+# Snake and Ladder Problem 
+
+Problem statement: Given that you have control on the outcome of the dice rolled in a snake and ladder game, find the minimum number of throws required to reach the end of the game. Below is the image of a snake and ladder board on which we have to find the number of throws.
+
+![snakeAndLadderBoard](assets\snakeAndLadderBoard.png)
+
+
+
+
+
+## How do we proceed further ?
+
+Here are the broad steps that one can keep in mind:
+
+ *  Convert the problem into a graph problem
+ *  The idea is to consider the given snake and ladder board as a directed
+    graph with number of vertices equal to the number of cells in the board
+ *  Since, all the edges are of "equal weight", it can be solved using breadth first search to find the shortest path
+
+
+
+

```

Comment:
@firekind @mitrish can we add more details about optimizations over here, like the approach where we discussed we tried to bring down the space complexity to 2n.

Reply:
This was left purposely, for other people to comment and optimise the solution